joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForStatementsCreator.scala

```diff
@@ -300,58 +300,27 @@ trait AstForStatementsCreator { this: AstCreator =>
     diffGraph.addEdge(localAstParentStack.head, iteratorLocalNode, EdgeTypes.AST)
     scope.addVariableReference(iteratorName, iteratorNode)
 
-    val callNode = createCallNode(
-      s"Object.keys($collectionName)[Symbol.iterator]()",
-      "",
-      DispatchTypes.DYNAMIC_DISPATCH,
-      forInOfStmt.lineNumber,
-      forInOfStmt.columnNumber
-    )
-
-    val thisNode = createIdentifierNode("this", forInOfStmt)
-
-    val indexCallNode = createCallNode(
-      s"Object.keys($collectionName)[Symbol.iterator]",
-      Operators.indexAccess,
+    val iteratorCall = createCallNode(
+      s"Runtime.iterator($collectionName)",
```

Comment:
this was my initial suggestion, but for consistency I think I'd prefer

```suggestion
      s"<operator>.iterator($collectionName)",
```